tests/MassTransit.KafkaIntegration.Tests/Avro_Specs.cs

```diff
@@ -22,21 +21,12 @@ public class Avro_Specs :
         [Test]
         public async Task Should_produce()
         {
-            static IAsyncSerializer<T> GetSerializer<T>(IServiceProvider provider)
-            {
-                return new AvroSerializer<T>(provider.GetService<ISchemaRegistryClient>());
-            }
-
-            static IDeserializer<T> GetDeserializer<T>(IServiceProvider provider)
-            {
-                return new AvroDeserializer<T>(provider.GetService<ISchemaRegistryClient>()).AsSyncOverAsync();
-            }
-
             await using var provider = new ServiceCollection()
                 .AddSingleton<ISchemaRegistryClient>(new CachedSchemaRegistryClient(new Dictionary<string, string>
                 {
                     { "schema.registry.url", "localhost:8081" },
                 }))
+                .AddSingleton<IKafkaSerializerFactory, AvroKafkaSerializerFactory>()
```

Comment:
If this is a singleton, it isn't bound to the bus/rider instance.

Reply:
well, it is in test how I did it, up to you how to register in your app